src/sirius/image.cc

```diff
@@ -43,14 +43,23 @@ Padding::Padding(int i_top, int i_bottom, int i_left, int i_right,
       right(i_right),
       type(i_type) {}
 
-Image::Image(const Size& size) : size(size) {
-    data.resize(size.CellCount());
-    std::fill(data.begin(), data.end(), 0);
-}
+Image::Image(const Size& size) : size(size), data(size.CellCount(), 0) {}
 
 Image::Image(const Size& size, Buffer&& buf)
     : size(size), data(std::move(buf)) {}
 
+Image::Image(Image&& rhs)
+    : size(std::move(rhs.size)), data(std::move(rhs.data)) {
+    rhs.size = {0, 0};
+}
+
+Image& Image::operator=(Image&& rhs) {
+    size = std::move(rhs.size);
```

Comment:
Note, since C++14, there is `std::exchange`: https://en.cppreference.com/w/cpp/utility/exchange

Reply:
Thanks, I did not know this function !